OpenRA.Mods.Common/Traits/World/MusicPlaylist.cs

```diff
@@ -84,6 +87,9 @@ public MusicPlaylist(World world, MusicPlaylistInfo info)
 				CurrentSongIsBackground = false;
 			}
 
+			if (info.DisableWorldSounds)
+				Game.Sound.DisableWorldSounds = true;
```

Comment:
It doesn't work in d2k for me and @obrakmann because we don't have any d2k music installed, so the `return` in line 62 prevents the code from ever getting here.

I guess you just need to move this check to the beginning of the constructor.